packages/ui/react-ui-syntax-highlighter/src/SyntaxHighlighter.tsx

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import React from 'react';
+import NativeSyntaxHighlighter, {
+  type SyntaxHighlighterProps as NativeSyntaxHighlighterProps,
+} from 'react-syntax-highlighter';
+// eslint-disable-next-line no-restricted-imports
+import styleDark from 'react-syntax-highlighter/dist/esm/styles/hljs/a11y-dark';
+// eslint-disable-next-line no-restricted-imports
+import styleLight from 'react-syntax-highlighter/dist/esm/styles/hljs/a11y-light';
+
+import { type ThemedClassName, useThemeContext } from '@dxos/react-ui';
+import { mx } from '@dxos/react-ui-theme';
+
+const zeroWidthSpace = '\u200b';
+
+export type SyntaxHighlighterProps = ThemedClassName<
+  NativeSyntaxHighlighterProps & {
+    fallback?: string;
+  }
+>;
+
+/**
+ * https://github.com/react-syntax-highlighter/react-syntax-highlighter
+ */
+export const SyntaxHighlighter = ({
```

Comment:
glad this is factored out, though fyi expecting we should deduplicate our syntax highlighting with the editor syntax highlighting https://github.com/dxos/dxos/issues/5301

Reply:
i considered that but i think it would be much heavier. esp for all of the testing/storybook/devtools use cases.

Reply:
Highlight.js is something that's jumped out every time I've done composer bundle reviews so I'm not sure what you mean by "heavier".